src/app/core/store/users/user.model.ts

```diff
@@ -22,7 +22,7 @@ import {GroupModel} from '../groups/group.model';
 export interface UserModel {
 
   id?: string;
-  name: string;
+  name?: string;
   email: string;
   groupsMap: { [organizationId: string]: string[]; };
   groups?: GroupModel[];
```

Comment:
I needed to convert groupsMap to usable format. So I added attribute here. Do you have any other idea how to do this?

Reply:
I think this is the perfect example of the distinction between DTOs and model classes. DTO needs to follow the structure provided by the back-end while our front-end entity model can use different structure which suits better our use-cases.

In this case, we can have `groups` field as a map in user DTO and `groups` field as an array in the user model. (Under the assumption that we always have only users for a single organization loaded on the front-end which is true as far as I know.) However, this field needs to be of type `string[]` and actual group instances should be loaded from the store by some specific selector.

The current solution is not very good because if you rename a group, this change will not be propagated to the users that you have already loaded on the front-end side.

Reply:
Oh, I see. In that case there is probably no need for a change here unless we come up with some better solution.

Reply:
well, we do not use groups yet so we can change it later..